resources/js/Components/Form/DmEntryCreateForm.tsx

```diff
@@ -0,0 +1,479 @@
+import {UsePageType} from 'Types/global'
+import {ItemData} from 'Types/item-data'
+import route from 'ziggy-js'
+
+type DmEntryCreateFormPropType = {
+    type: 'Edit' | 'Create'
+    onCloseDrawer?: () => void
+    editData?: EntriesData
+    editId?: number
+}
+type ItemDataType = {
+    name: string
+    description: string | null
+    rarity: string
+    tier: number
+}
+
+type DmEntryFormDataType = {
+    adventure_id: number
+    length: number
+    location: string
+    date_played: string | null
+    notes: string
+    items: ItemData[]
+    type: string
+    user_id: number | null | undefined
+}
+
+const StyledBox = styled(Box)`
+    padding: 32px 0px 16px 0px;
+`
+
+const FormBox = styled(Box)`
+    flexgrow: 1;
+    margin-top: 16px;
+`
+
+const StyledGrid = styled(Grid)`
+    align-items: center;
+    justify-content: center;
+    margin-bottom: 16px;
+`
+const StyledDateGrid = styled(Grid)`
+    margin-top: 32px;
+    margin-bottom: 23px;
+`
+
+const StyledFooter = styled(Grid)`
+    min-width: 25vw;
+`
+const StyledItemsFooter = styled(Grid)`
+    margin-top: 16px;
+    min-width: 25vw;
+`
+
+const DmEntryCreateForm = ({
+    type,
+    onCloseDrawer = () => {},
+    editData,
+    editId = 0,
+}: DmEntryCreateFormPropType) => {
+    const {auth, adventures} = usePage<UsePageType>().props
+    const {user} = auth
+    const RARITY = ['common', 'uncommon', 'rare', 'very_rare', 'legendary']
+    const ITEM_CREATE_FORM_INITIAL_VALUE: ItemDataType = {
+        name: '',
+        description: '',
+        rarity: '',
+        tier: 1,
+    }
+    const DM_ENTRY_CREATE_FORM_INITIAL_VALUE: DmEntryFormDataType = {
+        adventure_id: 0,
+        user_id: user?.id,
+        length: 0,
+        location: '',
+        date_played: new Date().toDateString(),
+        notes: '',
+        items: [],
+        type: 'dm',
+    }
+    const DM_ENTRY_FORM_INITIAL_VALUE: DmEntryFormDataType =
+        type === 'Create'
+            ? DM_ENTRY_CREATE_FORM_INITIAL_VALUE
+            : {
+                  adventure_id: editData?.adventure || 0,
+                  length: editData?.length || 0,
+                  location: editData?.location || '',
+                  date_played: editData?.date_played || new Date().toDateString(),
+                  notes: editData?.notes || '',
+                  items: editData?.items || [],
+                  type: 'dm',
+                  user_id: user?.id,
+              }
+
+    const {data, setData, errors, clearErrors, post, put} = useForm(
+        DM_ENTRY_FORM_INITIAL_VALUE,
+    )
+    const [activeStep, setActiveStep] = useState<number>(0)
+    const [item_data, setItemData] = useState<ItemData>({
+        name: '',
+        description: '',
+        rarity: '',
+        tier: 1,
+    })
+    const [items, setItems] = useState<ItemData[]>([])
+    const [isItemsVisible, setIsItemsVisible] = useState<boolean>(false)
+    const [error, setError] = useState<boolean>(false)
```

Comment:
error if i remember correctly should be server side response, hence it should be in props